test/e2e/storage/types/types.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
```

Comment:
What is the intention of moving ```TestSuite```, ```TestResource```, and ```TestDriver``` interface to new ```types``` package?

Can we just change ```TestDriver``` to handle ```DriverTestResources``` so that it can create and delete volume per test, and keep others almost as they are?
(I expect that only change for ```TestResource``` will be to get/pass ```DriverTestResources``` from/to ```TestDriver```.)

Reply:
There was some circular dependency issue that I had to deal with that prompted the move; however, I can't remember exactly what the issue was at this point. Is there are particular reason why we don't want a `types` package, it seems fairly idiomatic in the Kubernetes codebase.

Reply:
> There was some circular dependency issue that I had to deal with that prompted the move; however, I can't remember exactly what the issue was at this point.

I don't mean to stick to keep current code as is, so it's OK to move them to one place if there is no other way. However, just changing below way doesn't seem to introduce circular dependency issue.
(I only tested it with hostPathSymlink, so I'm not sure whether it works well in gce or other drivers as well.)

https://github.com/mkimuram/kubernetes/commit/5fc9bfa1db5d3709cef548fb05d9769eab320c73

So, I prefer to change only necessary part of the code to make it easier to review and understand the change in the logic. (And we will be able to move codes that way in a separate commit, if needed.)

>  Is there are particular reason why we don't want a types package, it seems fairly idiomatic in the Kubernetes codebase.

My intention of put them into the separate packages was to keep ```TestDriver``` and ```TestSuite``` loosely coupled as much as possible. To keep them loosely coupled will allow both driver developers and volume-feature developers focus almost only on their works when writing their e2e test codes.

Reply:
see #68569